src/main/java/com/minecolonies/coremod/entity/mobs/aitasks/RaiderRangedAI.java

```diff
@@ -100,7 +100,7 @@ protected void doAttack(final LivingEntity target)
         AbstractArrow arrowEntity = CombatUtils.createArrowForShooter(user);
 
         arrowEntity.setBaseDamage(user.getAttribute(MOB_ATTACK_DAMAGE).getValue());
-        if (flightCounter > 5)
+        if (flightCounter > 5 && arrowEntity instanceof CustomArrowEntity)
         {
             ((CustomArrowEntity) arrowEntity).setPlayerArmorPierce();
```

Comment:
Can start using named casts instead of separate `instanceof` and cast, unless you're planning to backport this to 1.16...